nupic/torch/functions/k_winners.py

```diff
@@ -99,11 +99,9 @@ def forward(ctx, x, duty_cycles, k, boost_strength):
         # Take the boosted version of the input x, find the top k winners.
         # Compute an output that contains the values of x corresponding to the top k
         # boosted values
-        res = torch.zeros_like(x)
         topk, indices = boosted.topk(k, sorted=False)
-        for i in range(x.shape[0]):
-            res[i, indices[i]] = x[i, indices[i]]
-
+        mask = torch.zeros_like(x).scatter(-1, indices, 1)
+        res = mask * x
         ctx.save_for_backward(indices)
```

Comment:
maybe we could save the `mask` instead of `indices` so we dont have to rebuild the mask in the backward call

Reply:
That makes sense. Can we assume that nothing else currently using this module accesses the saved indices?

Reply:
Yes. Saved indices is only used for backward pass.